firebase_admin/messaging.py

```diff
@@ -192,10 +250,72 @@ def __init__(self, code, message, detail=None):
         self.detail = detail
 
 
+class BatchResponse(object):
+
+    def __init__(self, responses):
+        if not isinstance(responses, list):
```

Comment:
I'm unsure how much validation is the right level of validation. Things seem different in different places in the project. Let me know and I can add/remove as necessary.